pkg/engine2/operational_eval/vertex_property_test.go

```diff
@@ -230,3 +231,136 @@ func Test_propertyVertex_Dependencies(t *testing.T) {
 		})
 	}
 }
+
+func Test_propertyVertex_evaluateConstraints(t *testing.T) {
+	id := construct.ResourceId{Provider: "test", Type: "test", Name: "test"}
+	type testData struct {
+		name    string
+		ref     construct.PropertyRef
+		res     *construct.Resource
+		rcs     []constraints.ResourceConstraint
+		mocks   func(mockProperty *MockProperty, mockRc *MockResourceConfigurer, data testData)
+		wantErr bool
+	}
+	tests := []testData{
+		{
+			name: "existing value and no constraints",
+			ref: construct.PropertyRef{
+				Resource: id,
+				Property: "test",
+			},
```

Comment:
nit: could use `graphtest.ParseRef` to let these be strings to be more compact without losing readability